chia/simulator/setup_nodes.py

```diff
@@ -349,77 +332,94 @@ async def setup_full_system_inner(
+            setup_vdf_clients(bt=b_tools, self_hostname=self_hostname, port=vdf1_port)
+        )
+        await async_exit_stack.enter_async_context(
+            setup_vdf_client(bt=shared_b_tools, self_hostname=self_hostname, port=vdf2_port)
+        )
+
+        daemon_ws = await async_exit_stack.enter_async_context(setup_daemon(btools=b_tools))
+
         # Start the introducer first so we can find out the port, and use that for the nodes
         introducer_service = await async_exit_stack.enter_async_context(setup_introducer(shared_b_tools, uint16(0)))
         introducer = introducer_service._api
         introducer_server = introducer_service._node.server
 
         # Then start the full node so we can use the port for the farmer and timelord
-        nodes = [
-            await async_exit_stack.enter_async_context(
-                setup_full_node(
-                    consensus_constants,
-                    f"blockchain_test_{i}.db",
-                    shared_b_tools.config["self_hostname"],
-                    b_tools if i == 0 else b_tools_1,
-                    introducer_server._port,
-                    False,
-                    10,
-                    True,
-                    connect_to_daemon=connect_to_daemon,
-                    db_version=db_version,
-                )
+        node_1 = await async_exit_stack.enter_async_context(
+            setup_full_node(
+                consensus_constants,
+                "blockchain_test_1.db",
+                self_hostname=self_hostname,
+                local_bt=b_tools,
+                introducer_port=introducer_server._port,
+                simulator=False,
+                send_uncompact_interval=0,
+                sanitize_weight_proof_only=False,
+                connect_to_daemon=connect_to_daemon,
+                db_version=db_version,
             )
-            for i in range(2)
-        ]
-        node_apis = [fni._api for fni in nodes]
-        full_node_0_port = node_apis[0].full_node.server.get_port()
+        )
+        node_2 = await async_exit_stack.enter_async_context(
+            setup_full_node(
+                consensus_constants,
+                "blockchain_test_2.db",
+                self_hostname=self_hostname,
+                local_bt=b_tools_1,
+                introducer_port=introducer_server._port,
+                simulator=False,
+                send_uncompact_interval=10,
+                sanitize_weight_proof_only=True,
+                connect_to_daemon=False,  # node 2 doesn't connect to the daemon
+                db_version=db_version,
+            )
+        )
+
         farmer_service = await async_exit_stack.enter_async_context(
             setup_farmer(
                 shared_b_tools,
                 shared_b_tools.root_path / "harvester",
-                shared_b_tools.config["self_hostname"],
-                consensus_constants,
-                full_node_0_port,
+                self_hostname=self_hostname,
```

Comment:
Does `shared_b_tools.config["self_hostname"]` have the wrong value?

Reply:
No I just used a variable at the top `self_hostname = shared_b_tools.config["self_hostname"]`